apps/examples/rtc_alarm/rtc_alarm_main.c

```diff
@@ -0,0 +1,353 @@
+
+	ret = sigaction(CONFIG_EXAMPLES_RTC_ALARM_SIGNO, &act, NULL);
+	if (ret < 0) {
+		int errcode = errno;
+		fprintf(stderr, "ERROR: sigaction failed: %d\n", errcode);
+		goto errout;
+	}
+
+	/* Now loop forever, waiting for alarm signals */
+
+	for (; ; ) {
+		/* Check if any alarms fired.
+		*
+		* NOTE that there are race conditions here... if we missing an alarm,
+		* we will just report it a half second late.
+		*/
+		for (i = 0; i < CONFIG_RTC_NALARMS; i++) {
+			if (g_alarm_received[i]) {
+				printf("alarm_daemon: alarm %d received\n", i);
+				g_alarm_received[i] = false;
+			}
+		}
+
+		/* Now wait a little while and poll again.  If a signal is received
+		* this should cause us to awken earlier.
+		*/
+		usleep(500 * 1000L);
+	}
+
+errout:
+	g_alarm_daemon_started = false;
+
+	printf("alarm_daemon: Terminating\n");
+	return EXIT_FAILURE;
+}
+
+/****************************************************************************
+ * Name: start_daemon
+ ****************************************************************************/
+
+static int start_daemon(void)
+{
+	if (!g_alarm_daemon_started) {
+		g_alarm_daemon_pid = task_create("alarm_daemon", CONFIG_EXAMPLES_RTC_ALARM_PRIORITY, CONFIG_EXAMPLES_RTC_ALARM_STACKSIZE, alarm_daemon, NULL);
+		if (g_alarm_daemon_pid < 0) {
+			int errcode = errno;
+			fprintf(stderr, "ERROR: Failed to start alarm_daemon: %d\n", errcode);
+			return -errcode;
+		}
+
+		printf("alarm_daemon started\n");
+		usleep(500 * 1000L);
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: show_usage
+ ****************************************************************************/
+
+static void show_usage(FAR const char *progname)
+{
+	fprintf(stderr, "USAGE:\n");
+	fprintf(stderr, "\t%s [-a <alarmid>] [-r <alarmid>] [-cr] [<seconds>]\n", progname);
+	fprintf(stderr, "Where:\n");
+	fprintf(stderr, "\t-a <alarmid> set alarm by absolute time\n");
+	fprintf(stderr, "\t\t<alarmid> selects the alarm: 0..%d (default: 0)\n", CONFIG_RTC_NALARMS - 1);
+	fprintf(stderr, "\t-r <alarmid> set alarm by relative time\n");
+	fprintf(stderr, "\t\t<alarmid> selects the alarm: 0..%d (default: 0)\n", CONFIG_RTC_NALARMS - 1);
+	fprintf(stderr, "\t-c\tCancel previously set alarm\n");
+	fprintf(stderr, "\t<seconds>\n");
+	fprintf(stderr, "\t\tThe number of seconds until the alarm expires.\n");
+	fprintf(stderr, "\t\t(only if no -c option given.)\n");
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * alarm_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int rtc_alarm_main(int argc, char *argv[])
+#endif
+{
```

Comment:
We dont need to make new app for this. Let us add this in rtc app.

Reply:
So, I have updated the code and added sample code to existing rtc app.